x/pki/handler_test.go

```diff
@@ -1462,6 +1510,45 @@ func TestHandler_RevokeX509Cert(t *testing.T) {
 	}
 }
 
+func TestHandler_RevokeX509RootCertsBySubjectKeyId(t *testing.T) {
+	setup := Setup(t)
+
+	// add root certificates
+	rootCertOptions := &rootCertOptions{
+		pemCert:      testconstants.PAACertWithSameSubjectID1,
+		subject:      testconstants.PAACertWithSameSubjectID1Subject,
+		subjectKeyID: testconstants.PAACertWithSameSubjectIDSubjectID,
+		info:         testconstants.Info,
+		vid:          testconstants.Vid,
+	}
+	proposeAndApproveRootCertificate(setup, setup.Trustee1, rootCertOptions)
+	rootCertOptions.pemCert = testconstants.PAACertWithSameSubjectID2
+	rootCertOptions.subject = testconstants.PAACertWithSameSubjectID2Subject
+	proposeAndApproveRootCertificate(setup, setup.Trustee1, rootCertOptions)
+
+	// revoke certificate
+	revokeX509Cert := types.NewMsgProposeRevokeX509RootCert(
+		setup.Trustee1.String(), testconstants.PAACertWithSameSubjectID1Subject, testconstants.PAACertWithSameSubjectIDSubjectID, testconstants.Info)
+	_, err := setup.Handler(setup.Ctx, revokeX509Cert)
+	require.NoError(t, err)
+
+	aprRevokeX509Cert := types.NewMsgApproveRevokeX509RootCert(
+		setup.Trustee2.String(), testconstants.PAACertWithSameSubjectID1Subject, testconstants.PAACertWithSameSubjectIDSubjectID, testconstants.Info)
+	_, err = setup.Handler(setup.Ctx, aprRevokeX509Cert)
+	require.NoError(t, err)
+
+	// check that root certificate has been revoked
+	approvedCertificates, _ := queryApprovedCertificates(setup, testconstants.PAACertWithSameSubjectID2Subject, testconstants.PAACertWithSameSubjectIDSubjectID)
+	require.Equal(t, 1, len(approvedCertificates.Certs))
+	require.Equal(t, testconstants.PAACertWithSameSubjectID2Subject, approvedCertificates.Certs[0].Subject)
+	require.Equal(t, testconstants.PAACertWithSameSubjectIDSubjectID, approvedCertificates.SubjectKeyId)
+
```

Comment:
I believe we need to query by SubjectKeyId here and make sure that one certificate is returned.